### DISCUSSION TITLE
Text Search returns error if there is a space in the string

### DISCUSSION THREAD

**Discussion by fergusmeiklejohn** [supabase/supabase#3519] [Questions] [answered]
If a user types a search phrase with a space like "Supabase Awesome", textSearch() will return an error.  This answer on stack overflow might be why: https://stackoverflow.com/a/16466057/5376269

I wrote this little function to build a search string from raw user input, which works okay in limited testing I've done:
```ts
type Operator = '&' | '|'

function buildTextSearchString(rawSearchString: string, operator: Operator) {
  const arrayOfWords = rawSearchString.trim().split(' ')
  let textSearchString = `'${arrayOfWords[0]}'`
  for (let i = 1; i < arrayOfWords.length; i++){
    textSearchString += ` ${operator} '${arrayOfWords[i]}'`
  }
  return textSearchString
}
```

Would it make sense to fold this function into `textSearch()`?

**Comment by bnjmnt4n** [comment, marked as answer]
See https://supabase.io/docs/reference/javascript/textsearch#full-normalization-1, which use `websearch_to_tsquery` to handle common web search style queries.

**Reply by fergusmeiklejohn** [reply]
thanks, yeah I just found this. It either should be in here or the link should be in here: https://supabase.io/docs/guides/database/full-text-search

**Reply by fergusmeiklejohn** [reply]
Overall the docs are difficult to navigate. There is a lot of duplication and focus on high level marketing type docs
